src/tasks/withdraw/safe.ts

```diff
@@ -0,0 +1,59 @@
+import SafeApiKit from "@safe-global/api-kit";
+import Safe, { EthersAdapter } from "@safe-global/protocol-kit";
+import { SafeTransactionDataPartial } from "@safe-global/safe-core-sdk-types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+interface Transaction {
+  authoringSafe: string;
+  to: string;
+  data: string;
+}
+
+function serviceUrlForNetwork(network: string): string {
+  if (["goerli", "mainnet", "gnosis-chain"].includes(network)) {
+    return `https://safe-transaction-${network}.safe.global`;
+  } else if (network === "xdai") {
+    return "https://safe-transaction-gnosis-chain.safe.global/";
+  } else {
+    throw new Error(`Unsupported network ${network}`);
+  }
+}
+
+// Creates and proposes a transaction to the Safe Multisig, which can then be confirmed by other signers in the Web UI. Returns the link to the transaction in the Web UI.
+export async function proposeTransaction(
+  { ethers }: HardhatRuntimeEnvironment,
+  network: string,
+  { authoringSafe, to, data }: Transaction,
+): Promise<string> {
+  const proposer = (await ethers.getSigners())[0];
+  const ethAdapter = new EthersAdapter({
+    ethers,
+    signerOrProvider: proposer,
+  });
+
+  const safeTransactionData: SafeTransactionDataPartial = {
+    to,
+    data,
+    value: "0",
+  };
+
+  const safeSdk = await Safe.create({ ethAdapter, safeAddress: authoringSafe });
+  const safeTransaction = await safeSdk.createTransaction({
+    safeTransactionData,
+  });
+  const safeTxHash = await safeSdk.getTransactionHash(safeTransaction);
+  const senderSignature = await safeSdk.signTransactionHash(safeTxHash);
+
+  const safeService = new SafeApiKit({
+    txServiceUrl: serviceUrlForNetwork(network),
+    ethAdapter,
+  });
+  await safeService.proposeTransaction({
+    safeAddress: authoringSafe,
+    safeTransactionData: safeTransaction.data,
+    safeTxHash,
+    senderAddress: await proposer.getAddress(),
+    senderSignature: senderSignature.data,
+  });
```

Comment:
This would require the private key of the proposer account to be put into `ethers`, right?
That is not doable long term as we are trying to move towards KMS which doesn't give us access to that data.
Maybe signing with the proposer could be done behind a feature flag for now?

Reply:
The Safe transaction service doesn't allow for unsigned transactions to be proposed (likely to prevent spam). This is why we created the "proposer account" in the first place. I think this account has significantly lower security assumptions than e.g. the solution submission accounts, so I'd be fine exporting the private key as a secret for this use case (otherwise I don't think we can automate the proposal).